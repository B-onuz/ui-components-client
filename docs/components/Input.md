# Input (Component)

Componente para entrada de texto.

## Como utilizar

```js
import Input from 'ui-components/lib/components/Input'
```

```jsx
<Input />
```

## Input controlada

```jsx
import React, { useState } from 'react'

const FooBar = () => {
  const [firstname, setFirstName] = useState('')
  return <Input id={'firstname'} name={'firstname'} placeholder={placeholder} value={firstname} onChange={(e) => setFirstName(e.target.value)} />
}

export default FooBar
```
