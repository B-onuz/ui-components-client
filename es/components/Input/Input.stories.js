'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.InputExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Input = _interopRequireDefault(require('./Input'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Input/Input.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Input',
  component: _Input['default'],
}
exports['default'] = _default

var InputExample = function InputExample() {
  return /*#__PURE__*/ _react['default'].createElement(_Input['default'], {
    placeholder: 'Nome inteiro',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 9,
      columnNumber: 35,
    },
  })
}

exports.InputExample = InputExample
