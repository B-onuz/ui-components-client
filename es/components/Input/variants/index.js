"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "size", {
  enumerable: true,
  get: function get() {
    return _size["default"];
  }
});

var _size = _interopRequireDefault(require("./size"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }