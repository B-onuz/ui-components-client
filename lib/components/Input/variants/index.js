"use strict";

exports.__esModule = true;

var _size = _interopRequireDefault(require("./size"));

exports.size = _size["default"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }