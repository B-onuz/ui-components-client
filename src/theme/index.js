const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
