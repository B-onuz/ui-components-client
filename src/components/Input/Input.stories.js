import React from 'react'
import Input from './Input'

export default {
  title: 'Input',
  component: Input,
}

export const InputExample = () => <Input placeholder="Nome inteiro" />
