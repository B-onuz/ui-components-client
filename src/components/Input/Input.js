import React from 'react'
import styled from 'styled-components'
import propTypes from '@styled-system/prop-types'
import { size } from './variants'
import Color from 'color'

const StyledInput = styled.input`
  border: 1px solid #dededf;
  border-radius: 4px;
  line-height: 2.1em;
  outline: none;
  padding-right: 1em;
  padding-left: 1em;
  width: 100%;
  box-sizing: border-box;
  &::placeholder {
    letter-spacing: 0px;
    color: #b5b5b5;
  }
  &:focus {
    box-shadow: 0 0 3px
      ${({ theme, color, error }) => (error ? new Color(theme.colors['red']).fade(0.5).string() : new Color(theme.colors[color]).fade(0.5).string())};
    border-color: ${({ theme, color, error }) => (error ? new Color(theme.colors['red']).string() : theme.colors[color])};
  }
  ${({ error, theme }) =>
    !!error &&
    `
      box-shadow: 0 0 3px ${new Color(theme.colors['red']).fade(0.5).string()};
      border: 1px solid ${theme.colors['red']};
    `}
  ${size}
`

const Input = ({ children, ...rest }) => {
  return <StyledInput {...rest} />
}

Input.propTypes = {
  ...propTypes.size,
}

Input.defaultProps = {
  size: 'medium',
  color: 'blue',
}

export default Input
