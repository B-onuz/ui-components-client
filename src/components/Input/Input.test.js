import React from 'react'
import Input from './Input.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render Input component', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Input id="my-input" placeholder="My input" />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
