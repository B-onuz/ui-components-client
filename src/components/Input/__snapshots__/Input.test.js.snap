// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Input component 1`] = `
.c0 {
  border: 1px solid #dededf;
  border-radius: 4px;
  line-height: 2.1em;
  outline: none;
  padding-right: 1em;
  padding-left: 1em;
  width: 100%;
  box-sizing: border-box;
  font-size: 16px;
}

.c0::-webkit-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0::-moz-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0:-ms-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0::placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0:focus {
  box-shadow: 0 0 3px rgba(33,133,208,0.5);
  border-color: #2185d0;
}

<input
  className="c0"
  color="blue"
  id="my-input"
  placeholder="My input"
  size="medium"
/>
`;
